source/Octopus.Versioning/Octopus/OctopusVersionMaskParser.cs

```diff
@@ -21,27 +21,27 @@ public class OctopusVersionMaskParser
             // Get the major version number
             @$"(?<{Major}>\d+|{PatternIncrement}|{PatternCurrent})" +
             // Get the minor version number, delimited by a period, comma, dash or underscore
-            @$"(?:[.\-_](?<{Minor}>\d+|{PatternIncrement}|{PatternCurrent}))?" +
+            @$"(?:\.(?<{Minor}>\d+|{PatternIncrement}|{PatternCurrent}))?" +
             // Get the patch version number, delimited by a period, comma, dash or underscore
-            @$"(?:[.\-_](?<{Patch}>\d+|{PatternIncrement}|{PatternCurrent}))?" +
+            @$"(?:\.(?<{Patch}>\d+|{PatternIncrement}|{PatternCurrent}))?" +
             // Get the revision version number, delimited by a period, comma, dash or underscore
-            @$"(?:[.\-_](?<{Revision}>\d+|{PatternIncrement}|{PatternCurrent}))?)?" +
+            @$"(?:\.(?<{Revision}>\d+|{PatternIncrement}|{PatternCurrent}))?)?" +
             // Everything after the last digit and before the plus is the prerelease
-            @$"(?:[.\-_])?(?<{Prerelease}>(?<{PrereleasePrefix}>[^+.\-_\s]*?)([.\-_](?<{PrereleaseCounter}>[^+\s]*?)?)?)?" +
+            @$"(?:-(?<{Prerelease}>(?<{PrereleasePrefix}>[^+.\-_\s]*?)([.\-_](?<{PrereleaseCounter}>[^+\s]*?)?)?)?)?" +
```

Comment:
This is the change that fixes the reported issue. It was here that a prerelease that started with an 'i' or 'c' was counted as a mask, where it wasn't previously.